CONTRIBUTING.md

```diff
@@ -1,3 +1,94 @@
 # Contributing
 
 Welcome! As a [Jupyter](https://jupyter.org) project, we follow the [Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
+
+Make sure to also follow [Project Jupyter's Code of Conduct](https://github.com/jupyter/governance/blob/master/conduct/code_of_conduct.md)
+for a friendly and welcoming collaborative environment.
+
+This guide was adapted from the [contributing guide in the main `jupyterhub` repo.](https://github.com/jupyterhub/jupyterhub/blob/master/CONTRIBUTING.md)
+
+## Setting up a development environment
+
+JupyterHub requires Python >= 3.5.
+
+As a Python project, a development install of JupyterHub follows standard practices for installation and testing.
+
+Note: if you have Docker installed locally, you can run all of the subsequent commands inside of a container after you run the following initial commands:
+
+```
+./ci/docker-ldap.sh
+docker run -v $PWD:/usr/local/src --workdir /usr/local/src --net=host --rm -it python:3.6 bash
+```
+
+1. Do a development install with pip
+
+    ```bash
+    cd ldapauthenticator
+    python3 -m pip install --editable .
+    ```
+
+    _Note: if you have Docker installed locally
```

Comment:
I think the end of this sentence got eaten by some hungry networking goblin?

Reply:
Heh I just forgot to remove my initial partial sentence after I moved it up to the top of the section 🤦‍♂